src/charm.py

```diff
@@ -421,138 +371,57 @@ def get_all_hostnames(self) -> List[str]:
-                )
-                peer_relation = self.model.get_relation("nginx-peers")  # type: ignore[arg-type]
-                self._tls.pop_relation_data_fields(
-                    [f"key-{hostname}", f"password-{hostname}"],
-                    peer_relation,  # type: ignore[arg-type]
-                )
-            except KeyError:
-                LOGGER.warning("Relation data for %s already does not exist", hostname)
-            self.certificates.request_certificate_revocation(
-                certificate_signing_request=old_csr.encode()
-            )
-
-    def _on_certificate_invalidated(self, event: CertificateInvalidatedEvent) -> None:
-        """Handle the TLS Certificate invalidation event.
-
-        Args:
-            event: The event that fires this method.
+        Returns:
+            The TLS certificates relation of the charm.
         """
-        tls_certificates_relation = self._tls.get_tls_relation()
-        if not tls_certificates_relation:
-            self.unit.status = WaitingStatus("Waiting for certificates relation to be created")
-            event.defer()
-            return
-        if event.reason == "revoked":
-            hostname = self._tls.get_hostname_from_cert(event.certificate)
-            self._certificate_revoked([hostname])
-        if event.reason == "expired":
-            self._tls.certificate_expiring(event, self.certificates)
-        self.unit.status = MaintenanceStatus("Waiting for new certificate")
+        relation = self.model.get_relation(TLS_CERT)
+        return relation
 
-    def _on_all_certificates_invalidated(self, _: AllCertificatesInvalidatedEvent) -> None:
-        """Handle the TLS Certificate relation broken event.
-
-        Args:
-            _: The event that fires this method.
-        """
-        tls_relation = self._tls.get_tls_relation()
-        if tls_relation:
-            tls_relation.data[self.app].clear()
-        if JujuVersion.from_environ().has_secrets:
-            hostnames = self.get_all_hostnames()
-            for hostname in hostnames:
-                try:
-                    secret = self.model.get_secret(label=f"private-key-{hostname}")
-                    secret.remove_all_revisions()
-                except SecretNotFoundError:
-                    LOGGER.warning("Juju secret for %s already does not exist", hostname)
+    def _get_certificate_requests(self) -> List[str]:
+        """Get the certificate requests for the charm."""
+        hostnames = self.get_all_hostnames()
+        return [
+            CertificateRequestAttributes(common_name=hostname, sans_dns=frozenset([hostname]))
+            for hostname in hostnames
+        ]
+
+    def _get_tls_certs(self) -> Dict[Union[str, None], Union[str, None]]:
+        """Get the TLS certificates provided by the TLS provider."""
+        certs: Dict[Union[str, None], Union[str, None]] = {}
+        if not hasattr(self, "certificates"):
+            return certs
```

Comment:
`_get_tls_certs` is called through the `get_all_hostnames` function which is called in the init in order to get the hostnames for which to request certificates. 

Ideally it would be possible to list the hostnames for which we want to request certificates without providing certs/keys but I didn't want to make changes to `IngressDefinitionEssence` or anything Ingress related